TheMET/TheMET/ViewControllers/CategoryViewController.swift

```diff
@@ -83,44 +98,107 @@ class CategoryViewController: UIViewController, UITableViewDelegate, UITableView
     }
     
     private func reloadCategory() {
-        self.contentStatus = .loaded([])
+        self.contentStatus = .loading
+        self.loadListOfArts { [weak self] arts in
+            guard let arts = arts else {
+                self?.contentStatus = .failed
+                return
+            }
+            self?.contentStatus = .loaded(arts)
+        }
     }
     
+    private func loadCellImage(cell: ArtViewCell, art: Art) {
+        cell.imageState = .loading
+        cell.tag = art.objectID
+        guard let imageURL =  art.primaryImage else {
+            cell.imageState = .failed
+            return
+        }
+        self.imageLoader.loadImage(urlString: imageURL) { image in
```

Comment:
make the `cell` weak in this completion and check it with `guard` as the first line

Reply:
+